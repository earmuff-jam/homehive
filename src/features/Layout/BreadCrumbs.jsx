import { HomeRounded } from "@mui/icons-material";
import { Box, Breadcrumbs, Stack, Typography } from "@mui/material";
import { useLocation, useNavigate } from "react-router-dom";

export default function BreadCrumbs({ currentRoute }) {
  const location = useLocation();
  const navigate = useNavigate();

  const pathnames = location.pathname.split("/").filter((x) => x);

  return (
    <Breadcrumbs aria-label="breadcrumb" className="no-print">
      <Box
        component="span"
        onClick={() => navigate("/")}
        sx={{
          color: "primary.main",
          cursor: "pointer",
          textDecoration: "none",
        }}
        role="link"
        tabIndex={0}
      >
        <Stack direction="row" spacing={1}>
          <HomeRounded fontSize="small" />
          <Typography variant="caption">Home</Typography>
        </Stack>
      </Box>

      {pathnames.map((value, index) => {
        const last = index === pathnames.length - 1;
        const to = `/${pathnames.slice(0, index + 1).join("/")}`;

        return last ? (
          <Stack
            key={to}
            direction="row"
            spacing={1}
            sx={{ color: "text.primary" }}
          >
            {currentRoute?.config?.breadcrumb?.icon}
            <Typography variant="caption">
              {currentRoute?.config?.breadcrumb?.value}
            </Typography>
          </Stack>
        ) : null;
      })}
    </Breadcrumbs>
  );
}
