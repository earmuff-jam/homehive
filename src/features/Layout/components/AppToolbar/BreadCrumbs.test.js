import React from "react";

import BreadCrumbs from "./BreadCrumbs";
import { HomeRounded } from "@mui/icons-material";
import { fireEvent, render, screen } from "@testing-library/react";

jest.mock("react-router-dom", () => ({
  useNavigate: () => jest.fn(),
  useLocation: () => ({ pathname: "/" }),
  MemoryRouter: ({ children }) => <div>{children}</div>,
}));

jest.mock("common/utils", () => ({
  HomeRouteUri: "/home",
}));

// currentRoute?.config?.breadcrumb?.icon
const mockCurrentRouteData = {
  config: {
    breadcrumb: {
      icon: <HomeRounded />,
      value: "Mohit Home",
    },
  },
};

describe("Breadcrumbs tests", () => {
  describe("Breadcrubms snapshot test", () => {
    it("renders correctly and matches snapshot", () => {
      const { asFragment } = render(<BreadCrumbs />);
      expect(asFragment()).toMatchSnapshot();
    });
  });
  describe("Breadcrumbs component test", () => {
    it("shows Home link", () => {
      render(<BreadCrumbs currentRoute={mockCurrentRouteData} />);
      expect(screen.getByText("Home")).toBeInTheDocument();
    });

    it("navigates to home when Home is clicked", () => {
      const mockNavigate = jest.fn();
      require("react-router-dom").useNavigate = () => mockNavigate;
      require("react-router-dom").useLocation = () => ({
        pathname: "/something",
      });

      render(<BreadCrumbs currentRoute={null} />);
      fireEvent.click(screen.getByRole("link"));
      expect(mockNavigate).toHaveBeenCalledWith("/home");
    });

    it("shows current breadcrumb text and icon", () => {
      require("react-router-dom").useLocation = () => ({
        pathname: "/dashboard",
      });

      const currentRoute = {
        config: {
          breadcrumb: {
            value: "Dashboard",
            icon: <span data-testid="icon">ðŸ“Š</span>,
          },
        },
      };

      render(<BreadCrumbs currentRoute={currentRoute} />);
      expect(screen.getByText("Dashboard")).toBeInTheDocument();
      expect(screen.getByTestId("icon")).toBeInTheDocument();
    });

    it("ignores intermediate paths", () => {
      require("react-router-dom").useLocation = () => ({
        pathname: "/admin/settings",
      });

      const currentRoute = {
        config: {
          breadcrumb: {
            value: "Settings",
          },
        },
      };

      render(<BreadCrumbs currentRoute={currentRoute} />);
      expect(screen.getByText("Settings")).toBeInTheDocument();
      expect(screen.queryByText("admin")).not.toBeInTheDocument();
    });

    it("does not render breadcrumb if currentRoute is missing", () => {
      require("react-router-dom").useLocation = () => ({ pathname: "/random" });

      render(<BreadCrumbs />);
      expect(screen.getByText("Home")).toBeInTheDocument();
      expect(screen.queryByText("random")).not.toBeInTheDocument();
    });
  });
});
