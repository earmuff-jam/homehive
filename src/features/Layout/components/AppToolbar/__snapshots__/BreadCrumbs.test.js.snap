// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Breadcrumbs tests Breadcrubms snapshot test renders correctly and matches snapshot 1`] = `
<DocumentFragment>
  <nav
    aria-label="breadcrumb"
    class="MuiTypography-root MuiTypography-body1 MuiBreadcrumbs-root no-print css-5cvx94-MuiTypography-root-MuiBreadcrumbs-root"
  >
    <ol
      class="MuiBreadcrumbs-ol css-1uwp4ue-MuiBreadcrumbs-ol"
    >
      <li
        class="MuiBreadcrumbs-li"
      >
        <span
          class="MuiBox-root css-17jq7r"
          role="link"
          tabindex="0"
        >
          <div
            class="MuiStack-root css-niqf4j-MuiStack-root"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeSmall css-120dh41-MuiSvgIcon-root"
              data-testid="HomeRoundedIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M10 19v-5h4v5c0 .55.45 1 1 1h3c.55 0 1-.45 1-1v-7h1.7c.46 0 .68-.57.33-.87L12.67 3.6c-.38-.34-.96-.34-1.34 0l-8.36 7.53c-.34.3-.13.87.33.87H5v7c0 .55.45 1 1 1h3c.55 0 1-.45 1-1"
              />
            </svg>
            <span
              class="MuiTypography-root MuiTypography-caption css-mpaqru-MuiTypography-root"
            >
              Home
            </span>
          </div>
        </span>
      </li>
    </ol>
  </nav>
</DocumentFragment>
`;
